app/main.py

```diff
@@ -1,8 +1,21 @@
 class Person:
-    # write your code here
-    pass
-
+    people = {}
+    def __init__(self, name: str, age: int):
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+    res_lis = [Person(person["name"], person["age"]) for person in people]
+
+    for per in people:
+        cur_per = Person.people[per["name"]]
+        if "wife" in per and per["wife"]:
+            cur_per.wife = Person.people[per["wife"]]
+        elif "husband" in per and per["husband"]:
+            cur_per.husband = Person.people[per["husband"]]
```

Comment:
When assigning a spouse, the code assumes that the spouse's name exists in the `Person.people` dictionary. Consider adding error handling to manage cases where the spouse is not found, which could occur if the input list is incomplete or contains errors.